jgrapht-core/src/main/java/org/jgrapht/alg/decomposition/NiceDecompositionBuilder.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * (C) Copyright 2018-2018, by Ira Justus Fesefeldt and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.decomposition;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.util.*;
+import org.jgrapht.graph.*;
+
+/**
+ * An abstract builder class for nice tree decompositions, which builds the tree decomposition in a
+ * top-down manner.
+ * <p>
+* A tree-decomposition of a graph $G = (V, E)$ is a pair $(X, T) = (\{X_i\ |\ i\in I\}, (I, F))$ where
+ * $X = \{X_i\ |\ i \in I \}$ is a family of subsets of $V$, and $T = (I, F)$ is a tree, such that
+ * <ul>
+ * <li>Union of the sets $X_i$ equals to $V$</li>
+ * <li>for every edge $e = (u,v)$ there exists a set $X_i$ such that $u \in X_i$ and $v \in X_i$</li>
+ * <li>if both $X_i$ and $X_j$ contain a vertex $v$ then every set $X_k$ on the simple path from $X_i$ to $X_j$
+ * contains vertex $v$.</li>
+ * </ul>
+ * <br>
```

Comment:
Let's mention the mapping `b` or replace its usages below with the sets X_i. I would choose the first alternative since it is convenient to think of the decomposition tree as the mapping from tree nodes to the sets containing vertices from the initial graph.